docs/source/concepts/recipe.rst

```diff
@@ -86,6 +92,115 @@ The `conda skeleton <skeleton_ref>` command can help to make
 skeleton recipes for common repositories, such as `PyPI
 <https://pypi.python.org/pypi>`_.
 
+
+Deep dive
+=========
+
+Let's take a closer look at how conda build uses a recipe
+to create a package.
+ 
+Templates
+---------
+
+When you build a conda package, conda build renders the package
+by reading a template in the meta.yaml. See :ref:`jinja-templates`.
+
+Templates are filled in using your conda build config,
+which shows the matrix of things to build against. The
+``conda build config`` determines how many builds it has to do.
+For example, defining a conda_build_config.yaml of the form
+and filling it defines a matrix of 4 packages to build::
+
+   python:
+     - 3.7
+     - 2.7
+   vs:
+     - 2008
+     - 2017
```

Comment:
This isn't really the best example, because these specific keys are coupled in practice.  I think it's mostly fine, but it might cause a bit of confusion for people who know too much to be dangerous.

Reply:
Is there a better example we can put in here instead?

Reply:
Maybe just a totally generic one:

```
foo:
  - 1.0
  - 2.0
bar:
  - 1.2.0
  - 1.4.0
```